### PR TITLE
Castform and Cherrim Single frame to Two frame additions, Other Graphical tweaks

### PR DESCRIPTION
**Pull request by SonikkuA-DatH**
This PR Adds a second frame for the following

- Castform and related forms
- Spinda
- Cherrim Sunshine

In addition, this PR also;

- Tweaks Castform and Spinda palettes
- Implements the Gen 8 Pokemon Home colors for Castform form shiny colors
- Makes Spheal spin again
- Enables Spinda to have spots for its second frame (special thanks to shinydragonhunter for that)

Gifs:
Castform CastformRainsped
CastformSnow CastformSun
CherrimSun Sphealsped
Spindasped

Discord ID: dat.ha, shinydragonhunter

### CODE REVIEW

File: src/data/pokemon_graphics/front_pic_anims.h [hunk 1]
```diff
@@ -4144,27 +4154,45 @@ static const union AnimCmd sAnim_Aggron_1[] =
...
+static const union AnimCmd sAnim_Castform_1[] =
 {
-    ANIMCMD_FRAME(0, 1),
-    ANIMCMD_END,
+	ANIMCMD_FRAME(0, 12),
+	ANIMCMD_FRAME(1, 24),
+	ANIMCMD_FRAME(0, 24),
+	ANIMCMD_FRAME(1, 24),
+	ANIMCMD_FRAME(0, 24),
+	ANIMCMD_END,
```
Comment (AsparagusEduardo) [inline src/data/pokemon_graphics/front_pic_anims.h:4164]: Spaces instead of tabs

File: src/data/pokemon_graphics/front_pic_anims.h [hunk 2]
```diff
@@ -4144,27 +4154,45 @@ static const union AnimCmd sAnim_Aggron_1[] =
...
+static const union AnimCmd sAnim_CastformSunny_1[] =
 {
-    ANIMCMD_FRAME(1, 1),
+	ANIMCMD_FRAME(0, 12),
+	ANIMCMD_FRAME(1, 12),
+	ANIMCMD_FRAME(0, 12),
+	ANIMCMD_FRAME(1, 12),
+	ANIMCMD_FRAME(0, 12),
+	ANIMCMD_FRAME(1, 12),
+	ANIMCMD_FRAME(0, 12),
```
Comment (AsparagusEduardo) [inline src/data/pokemon_graphics/front_pic_anims.h:4175]: Spaces instead of tabs

File: src/data/pokemon_graphics/front_pic_anims.h [hunk 3]
```diff
@@ -4144,27 +4154,45 @@ static const union AnimCmd sAnim_Aggron_1[] =
...
+static const union AnimCmd sAnim_CastformRainy_1[] =
 {
-    ANIMCMD_FRAME(2, 1),
-    ANIMCMD_END,
+	ANIMCMD_FRAME(0, 22),
+	ANIMCMD_FRAME(1, 22),
+	ANIMCMD_FRAME(0, 22),
+	ANIMCMD_FRAME(1, 22),
+	ANIMCMD_FRAME(0, 11),
+	ANIMCMD_END,
```
Comment (AsparagusEduardo) [inline src/data/pokemon_graphics/front_pic_anims.h:4186]: Spaces instead of tabs

File: src/data/pokemon_graphics/front_pic_anims.h [hunk 4]
```diff
@@ -4144,27 +4154,45 @@ static const union AnimCmd sAnim_Aggron_1[] =
...
-static const union AnimCmd sAnim_Castform_3[] =
+static const union AnimCmd sAnim_CastformSnowy_1[] =
 {
-    ANIMCMD_FRAME(3, 1),
+    ANIMCMD_FRAME(0, 5),
+	ANIMCMD_FRAME(1, 12),
+	ANIMCMD_FRAME(0, 12),
+	ANIMCMD_FRAME(1, 29),
+	ANIMCMD_FRAME(0, 12),
     ANIMCMD_END,
```
Comment (AsparagusEduardo) [inline src/data/pokemon_graphics/front_pic_anims.h:4196]: spaces instead of tabs

File: src/pokemon.c [hunk 5]
```diff
@@ -6795,97 +6800,92 @@ u16 HoennToNationalOrder(u16 hoennNum)
...
+				}
+
+				spotPixelRow >>= 1;
+			}
+
+			y++;
+		}
+
+		personality >>= 8;
+	}
```
Comment (AsparagusEduardo) [inline src/pokemon.c:6888]: spaces instead of tabs

File: src/pokemon.c [hunk 6]
```diff
@@ -8096,13 +8096,9 @@ const u8 *GetTrainerNameFromId(u16 trainerId)
...
-    return (species != SPECIES_CASTFORM
-         && species != SPECIES_SPINDA
-         && species != SPECIES_UNOWN
-         && species != SPECIES_CHERRIM
-         && species != SPECIES_CASTFORM_SUNNY
-         && species != SPECIES_CASTFORM_RAINY
-         && species != SPECIES_CASTFORM_SNOWY);
+    return (
+         species != SPECIES_UNOWN
+        );
```
Comment (AsparagusEduardo) [inline src/pokemon.c:8101]:
```suggestion
    return species != SPECIES_UNOWN;
```

File: src/pokemon.c [hunk 7]
```diff
@@ -6795,97 +6800,92 @@ u16 HoennToNationalOrder(u16 hoennNum)
...
+	the bit is set then it's part of the spot; try to draw it. A pixel is drawn on Spinda if the
+	pixel on Spinda satisfies the following formula: ((u8)(colorIndex - 1) <= 2). The -1 excludes
+	transparent pixels, as these are index 0. Therefore only colors 1, 2, or 3 on Spinda will
+	allow a spot to be drawn. These color indexes are Spinda's light brown body colors. To create
+	the spot it adds 4 to the color index, so Spinda's spots will be colors 5, 6, and 7.
+
+	The above is done two different ways in the function: one with << 4, and one without. This
+	is because Spinda's sprite is a 4 bits per pixel image, but the pointer to Spinda's pixels
+	(destPixels) is an 8 bit pointer, so it addresses two pixels. Shifting by 4 accesses the 2nd
+	of these pixels, so this is done every other time.
```
Comment (AsparagusEduardo) [inline src/pokemon.c:6825]: spaces instead of tabs

File: src/decompress.c [hunk 8]
```diff
@@ -114,7 +114,11 @@ void LoadSpecialPokePic(void *dest, s32 species, u32 personality, bool8 isFrontP
             LZ77UnCompWram(gMonBackPicTable[species].data, dest);
     }
 
-    DrawSpindaSpots(species, personality, dest, isFrontPic);
+	if (species == SPECIES_SPINDA && isFrontPic)
+	{
+		DrawSpindaSpots(personality, dest, FALSE);
+		DrawSpindaSpots(personality, dest, TRUE);
+	}
```
Comment (AsparagusEduardo) [inline src/decompress.c:121]: Spaces instad of tabs

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
- These look good but please, please, ***please*** don't use GitHub's file upload because it messes with linebreak stuff:
imagen
- Please update Cherrim's palette in the image to match the in-game palette/removed front.png
- Please use the official 2nd frame for Cherrim provided by CyanSixFour.
- What did you change about Spheal's sprite?

**Comment by SonikkuA-DatH** [comment]
> * These look good but please, please, _**please**_ don't use GitHub's file upload because it messes with linebreak stuff:
>   imagen
> * Please update Cherrim's palette in the image to match the in-game palette/removed front.png
> * Please use the official 2nd frame for Cherrim provided by CyanSixFour.
> * What did you change about Spheal's sprite?

Huh, pal's break that badly for upload?
Spheal I accidentally replaced with Spinda in dev. I immediately fixed that, though tweaked mouth to better match perspective. In initial expansion it uses V stretch for front anim (despite still using vanilla anim timings and delay). Visually user end only the spin matters
Will fix Cherrim and spacing soon

**Pull request closed without merging**